libimage/runtime_test.go

```diff
@@ -81,7 +80,7 @@ func TestTmpdir(t *testing.T) {
 	require.Equal(t, "/tmp/from/containers.conf", tmpd)
 
 	if confSet {
-		os.Setenv(confStr, conf)
+		t.Setenv(confStr, conf)
 	} else {
 		os.Unsetenv(confStr)
 	}
```

Comment:
Using a nested `t.Run()` with a `t.Setenv()` in the inner function should allow us to remove the manual `confSet` cleanup. (I verified that `t.Cleanup()` for the nested function is run before `t.Run()` returns.)

Reply:
yeah honestly I had no idea what this is complicated mess tried to achieve really.

I think you are right these are to test cases and with nesting it should work, I will try that